documentation/src/main/asciidoc/topics/con_jvm_heap_memory_allocation.adoc

```diff
@@ -1,26 +1,127 @@
+
+[source,options="nowrap",subs=”+attributes.+quotes”]
+----
+JVM memory = JVM code + JVM stack size + JVM heap space
+----
+
+Note the following elements from the previous calculation:
+* A JVM stack, also called a thread stack, is an area of JVM memory that stores data that can be used by a JVM thread. 
+* JVM heap space is an area of JVM memory that stores data and temporary objects from an application. 
+
+Additionally, consider the following calculation that can help you determine a suitable JVM heap space, where `m1` represents the calculated JVM heap space in megabytes. 
+
+[source,options="nowrap",subs=”+attributes.+quotes”]
+----
+m1 = (int) ((x * y) / (p * S)) + 1
+----
+
+The following table outlines the elements in the previous calculation:
+
+[%autowidth,cols="1,1",stripes=even,%noheader]
+|===
+| Element | Description
+
+|`int` | Denotes the calculated value for `m1` as an integer. 
+| `x` | Represents the average size of a key-value pair. You can determine the value of `x` bu using the following calculator: `x =  Serialized size of key + Serialized size of value + Metadata`
+| `y` a| Represents the expected number of cache entries for each node. Consider the type of cache mode when determining a value for `y`. For example:
+
+* For distributed mode, determine the volume of owners per cache entry.
+* For the local and replicated modes, 
+
+Note: Ensure that the number of entries (H) does not exceed the expected JVM heap space (S). Otherwise, {brandname} might evict more cache entries to optimize JVM memory space
+
+| `p` | 
+Denotes a fractional value for the full heap space that Infinsipan uses to store cache entries. Typical starting value is p=0.5.
+
+Consider that the JVM, heap size, GC parameters, and the volume of data stored in JVM memory can affect the length of a GC operation. You might need to configure your GC implementation to facilitate these operations.
+
+| `S` | 
+Represents the expected JVM heap space for each node in a cluster. Value in megabytes.
+|===
+
+////
+
 .Off-heap storage
 
-{brandname} uses JVM heap representations of objects to process read and write operations on caches or perform other operations such as state transfer.
+{brandname} uses JVM heap representations of objects to process read and write operations on caches or perform other operations, such as a state transfer operation.
 You must always allocate some JVM heap space to {brandname}, even if you store entries in off-heap memory.
 
-The amount of JVM heap memory that {brandname} uses with off-heap storage is much smaller than when storing data in the JVM heap space.
-However that amount increases as the number of concurrent operations scales up.
+The volume of JVM heap memory that {brandname} uses with off-heap storage is much smaller when compared with storing data in the JVM heap space.
+The JVM heap memory increases as the number of concurrent operations scales up.
```

Comment:
Maybe something more like "The JVM heap memory requirements for off-heap storage scale with the number of concurrent operations instead of number of stored entries" or something like that.

Reply:
Thanks. Updated.